src/views/manage-identities/ManageIdentities.tsx

```diff
@@ -0,0 +1,403 @@
+    mutationFn: async () => {
+      return writeIdentityRegistryUpdateIdentity(
+        walletInterface as WalletInterface,
+        {
+          args: [
+            selectedIdentity as EvmAddress,
+            newIdentityAddress as EvmAddress,
+          ],
+        },
+        registry as EvmAddress,
+      );
+    },
+  });
+
+  const { mutateAsync: mutateIdentityRegistryRemoveAgent } = useMutation({
+    mutationFn: async () => {
+      return writeIdentityRegistryRemoveAgent(
+        walletInterface as WalletInterface,
+        { args: [selectedAgent as EvmAddress] },
+        registry as EvmAddress,
+      );
+    },
+  });
+
+  const { mutateAsync: mutateIdentityRegistryAddAgent } = useMutation({
+    mutationFn: async () => {
+      return writeIdentityRegistryAddAgent(
+        walletInterface as WalletInterface,
+        { args: [newUserAgentAddress as EvmAddress] },
+        registry as EvmAddress,
+      );
+    },
+  });
+
+  const addNewUserAgent = async () => {
+    if (ethers.isAddress(newUserAgentAddress)) {
+      try {
+        const txHash = await mutateIdentityRegistryAddAgent();
+        setUpdateTxResult(txHash);
+        setUpdateTxError(undefined);
+      } catch (err: any) {
+        setUpdateTxResult(undefined);
+        setUpdateTxError(err);
+      }
+    } else {
+      setUpdateTxError("User agent address is incorrect");
+      setUpdateTxResult(undefined);
+    }
+  };
+
+  const handleTokenSelect = (tokenAddr: string) => {
+    setSelectedToken(
+      ownTokens.find((token) => token.address === (tokenAddr as string)),
+    );
+  };
+
+  const handleIdentitySelect = (value: string) => {
+    setSelectedIdentity(value as `0x${string}`);
+  };
+
+  const handleUpdateIdentity = async () => {
+    if (ethers.isAddress(newIdentityAddress)) {
+      try {
+        const txHash = await mutateUpdateIdentity();
+        setUpdateTxResult(txHash);
+        setUpdateTxError(undefined);
+      } catch (err: any) {
+        setUpdateTxError(err);
+        setUpdateTxResult(undefined);
+      }
+    } else {
+      setUpdateTxError("New identity address is incorrect");
+      setUpdateTxResult(undefined);
+    }
+  };
+
+  const handleDeleteIdentity = async () => {
+    try {
+      const txHash = await mutateDeleteIdentity();
+      setUpdateTxResult(txHash);
+      setUpdateTxError(undefined);
+    } catch (err: any) {
+      setUpdateTxError(err);
+      setUpdateTxResult(undefined);
+    }
+  };
+
+  const handleUpdateIdentityCountry = async (value: string | number) => {
+    try {
+      const txHash = await mutateUpdateIdentityCountry({
+        country: value as number,
+      });
+      setUpdateTxResult(txHash);
+      setUpdateTxError(undefined);
+    } catch (err: any) {
+      setUpdateTxError(err);
+      setUpdateTxResult(undefined);
+    }
+  };
+
+  const handleAgentSelect = (agent: string | number) => {
+    setSelectedAgent(agent as `0x${string}`);
```

Comment:
ideally we would drop the `as` completely if possible, otherwise lets use `as EvmAddress` across all the places instead of `0x${string}`